### PR TITLE
added more flexibility to add a paragraph to a log area 

### PR DESCRIPTION
**Pull request by coldwarrl**
(possibility for styling etc.)

### CODE REVIEW

File: zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/impl/DefaultLogArea.kt [hunk 1]
```diff
@@ -116,7 +123,7 @@ class DefaultLogArea constructor(componentMetadata: ComponentMetadata,
             element.moveTo(Position.zero().relativeToBottomOf(lastChild))
         }
         addComponent(element)
-        element.applyColorTheme(currentTheme)
+        //element.applyColorTheme(currentTheme)
```
Comment (adam-arold) [inline zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/impl/DefaultLogArea.kt]: Why did you comment this?

File: zircon.core/src/main/kotlin/org/hexworks/zircon/api/builder/component/TextBoxBuilder.kt [hunk 2]
```diff
@@ -66,6 +67,16 @@ data class TextBoxBuilder(
         }
     }
 
+    @JvmOverloads
+    fun addParagraph(paragraph: Paragraph, withNewLine: Boolean = true) = also {
```
Comment (adam-arold) [inline zircon.core/src/main/kotlin/org/hexworks/zircon/api/builder/component/TextBoxBuilder.kt]: So I was thinking whether passing a `Builder<Paragraph>` would be better than passing an actual `Paragraph` instance. The reason for this is if the user passes a builder, we have a chance to fix its dimensions, but with a concrete `Paragraph` we do not.

**Comment by coldwarrl** [comment]
changed parapgrah to paragraphBuilder

**Comment by adam-arold** [comment]
Codacy is crying because of the undocumented public function. Would you kindly add a kdoc, then I can merge this? 👍

**Pull request merged**